### PR TITLE
Keep warm data : Add end2end tests

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

This PR add end2end tests for the [How-To Keep only warm data](https://github.com/kuzzleio/kuzzle-how-to/tree/master/keep-only-warm-data).  
The tests are realized with Cucumber ruby : 
 - Check if Kuzzle stack is running
 - Check the data loading in the past
 - Check the suppression of documents older than 30 days

The tests also include a lint of the scripts.

### How should this be manually tested?

  - Step 1 : Go to the `keep-only-warm-data` folder
  - Step 2 : Run `./run_tests.sh`

### Other changes

How-To#Try It Yourself updated with a script to count inserted and then deleted data. (`scripts/countData.js`)

### Boyscout
 
  - Remove the `docker-compose pull` before hook from cucumber and put it in `run_tests.sh` because we may want to test local images in the dev process

#### How-To Sync data to another database

 - Raise `StandardError` instead of `ArgumentError` when test fail
 - Use Kuzzle camelCase convention for script name
 - Fix Dockerfiles and docker-compose with @benoitvidis suggestions [here](https://github.com/kuzzleio/kuzzle-how-to/pull/4)

### CODE REVIEW

File: keep-only-warm-data/features/step_definitions/keep_warm_data.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    end
+  end
+
+  if ! connected
+    puts docker_compose_stderr.read_nonblock(99999)
+    puts docker_compose_stdout.read_nonblock(99999)
+    raise StandardError, "Unable to start docker-compose stack"
+  end
+
+  puts "Kuzzle is up !"
```
Comment (scottinet) [inline keep-only-warm-data/features/step_definitions/keep_warm_data.rb]: no space before an exclamation mark: `Kuzzle is up!`

File: keep-only-warm-data/scripts/loadData.js [hunk 2]
```diff
@@ -1,32 +1,27 @@
...
-  index = 'nyc-open-data';
+  port = 9200;
 
 let
   inserted = 0,
   headerSkipped = false,
   currentDay = 0;
 
-let currentDate = Moment().subtract(currentDay, 'days');
+let currentDate = moment().subtract(currentDay, 'days');
```
Comment (scottinet) [inline keep-only-warm-data/scripts/loadData.js]: use `const` instead of `let`

File: keep-only-warm-data/scripts/loadData.js [hunk 3]
```diff
@@ -1,32 +1,27 @@
...
 
 let documents = [];
```
Comment (scottinet) [inline keep-only-warm-data/scripts/loadData.js]: ditto

File: keep-only-warm-data/scripts/loadData.js [hunk 4]
```diff
@@ -79,37 +74,35 @@ dataFile.on('close', () => {
...
+function createPacket(docs) {
   const kuzzle_meta = {
-    "_kuzzle_info": {
-      "active":     true,
-      "author":     "-1",
-      "updater":    null,
-      "updatedAt":  null,
-      "deletedAt":  null,
-      "createdAt":  currentDate.valueOf() // Date in milli timestamp
+    '_kuzzle_info': {
```
Comment (scottinet) [inline keep-only-warm-data/scripts/loadData.js]: or better yet: no quote at all :open_mouth:

File: sync-data-to-another-database/docker/cassandra/Dockerfile [hunk 5]
```diff
@@ -1,7 +1,9 @@
 FROM cassandra:3
 
-ADD create_keyspace.cql docker-entrypoint-initdb.d/create_keyspace.cql
+LABEL maintainer="amaret@kuzzle.io"
```
Comment (scottinet) [inline sync-data-to-another-database/docker/cassandra/Dockerfile]: official kuzzle maintainer: `support@kuzzle.io`

File: sync-data-to-another-database/docker/kuzzle/Dockerfile [hunk 6]
```diff
@@ -1,7 +1,9 @@
 FROM kuzzleio/kuzzle:develop
 
-ADD custom_start.sh /custom_start.sh
+LABEL maintainer="amaret@kuzzle.io"
```
Comment (scottinet) [inline sync-data-to-another-database/docker/kuzzle/Dockerfile]: ditto

File: sync-data-to-another-database/plugin/manifest.json [hunk 7]
```diff
@@ -2,5 +2,5 @@
   "name": "kuzzle-plugin-sync-cassandra",
   "version": "1.2.0",
   "threadable": false,
-  "kuzzleVersion": "^1.1.x"
+  "kuzzleVersion": "^1.1.11"
```
Comment (scottinet) [inline sync-data-to-another-database/plugin/manifest.json]: `1.2.11`?

Comment (Aschen) [reply sync-data-to-another-database/plugin/manifest.json]: We need the new `m*` routes if we want to insert batch of 50 000 documents

**Pull request merged**